src/server.ts

```diff
@@ -197,9 +200,54 @@ connection.onDidChangeConfiguration((change) => {
 	}
 });
 
+function setSchemaStoreSettingsIfNotSet(){
+	if(schemaStoreSettings.length === 0){
+		getSchemaStoreMatchingSchemas().then(schemaStore => {
+			schemaStoreSettings = schemaStore.schemas;
+			updateConfiguration();
+		});
+	}
+}
+
+function getSchemaStoreMatchingSchemas(){
+	
+	return xhr({ url: "http://schemastore.org/api/json/catalog.json" }).then(response => {
+		
+		let languageSettings= {
+			schemas: []
+		};
+
+		let schemas = JSON.parse(response.responseText);
+		for(let schemaIndex in schemas.schemas){
+			
+			let schema = schemas.schemas[schemaIndex];
+			if(schema && schema.fileMatch){
+
+				for(let fileMatch in schema.fileMatch){
+					let currFileMatch = schema.fileMatch[fileMatch];
+
+					if(currFileMatch.indexOf('.yml') !== -1 || currFileMatch.indexOf('.yaml') !== -1){
+						languageSettings.schemas.push({ uri: schema.url, fileMatch: [currFileMatch] });
+					}
+
+				}
+				
+			}
+
+		}
+
+		return languageSettings;
+
+	}, (error: XHRResponse) => {
+		return Promise.reject(error.responseText || getErrorStatusDescription(error.status) || error.toString());

```

Comment:
inside a Promise callback you can use `throw` instead of `return Promise.reject()`
It's a good idea to always throw/reject `Error` instances instead of strings because strings don't have a stack trace and cannot have additional metadata.